forte/data/ontology/top.py

```diff
@@ -1052,12 +1062,35 @@ def __init__(self, pack: PackType, image_payload_idx: int = 0):
         else:
             self._image_payload_idx = image_payload_idx
 
+    @property
+    def image_payload_idx(self):
+        r"""Getter function of ``image_payload_idx``. The function will first try to
+        retrieve the image_payload_idx index from ``DataStore`` in ``self.pack``. If
+        this attempt fails, it will directly return the value in ``_image_payload_idx``.
+        """
+        try:
+            self._image_payload_idx = self.pack.get_entry_raw(self.tid)[
+                PAYLOAD_INDEX
```

Comment:
Let's just try not to add additional consts here. DataStore has a system to add attributes and datastore can manage the attribute position. Can we simply consider `Payload_index` to be a regular attribute?

Reply:
Yes. Every `Entry` has two main attributes which occupy the first and second position in the datastore (for Annotation it is begin and end index). Similar, for `ImageAnnotation` and its subclasses, the two main attributes are payload_idx and None (Implying that it only has one main attribute, the extra None is just for consistency). That is the reason I stored the Payload index value in `constants`

Reply:
what is the `None` here?

Reply:
> So Payload Index is the only main attribute for `ImageAnnotation` type entries (just like how `Annotation` has `start` and `end` indices, `payload_index` is stored in the first two indices of the `DataStore` entry. That's why I felt that could be the only constant we could add for `ImageAnnotation`

Ok, so I think the reason I have the confusion is that: once `Payload` is in our system, then all other payload-dependent entries (specifically, Annotation, ImageAnnotation, AudioAnnotation) should have `Payload` associated with it too. Now we only modify it for `ImageAnnotation` in this PR, it feels a bit inconsistent to me.

Reply:
1. So we should make all the three have a `Payload` field. I am not sure of what `data class attribute` means yet, but we need to make all of the consistent, but with two considerations:
    i. maybe it will make this PR too large?
    ii.  we don't want to change any existing behavior, i.e., previously we don't have multiple payloads for each modality, thus all entries are retrieved without any payload index. To keep the same behavior, we should assume any data operation that does not use a `payload index` default to 0, i.e., get from the first payload or add to the first payload when payload index is not specified.
2. Can you remind me what are `two main attributes`? I just wonder why it could be `None`.